metabase/metabase
# [MLv2] Add wrappers for column type checks, add `getColumnIcon` helper

Epic #28034

Adds TypeScript wrappers for column type helpers like `isPK`, `isNumeric`, etc. (based on #30054)

Also, fixes an issue that the column picker for order-by clauses stopped rendering icons correctly. Fixed by adding an MLv2-based `getColumnIcon` function that will replace [`getIconForField`](https://github.com/metabase/metabase/blob/442080089ddcea940e1a861f46802ac13c5a1f9f/frontend/src/metabase-lib/metadata/utils/fields.ts#L28). The new function works with opaque column objects and doesn't need explicit `base_type`, `effective_type`, and `semantic_type` 🎉 

### How to Verify

1. New > Question > Sample Database > Orders
2. Try to add a sort
3. Ensure each column in the column picker has an icon

### Demo

| Before | After  |
|--------|-------|
| <img width="385" alt="Before" src="https://user-images.githubusercontent.com/17258145/235486520-1d6fb8f8-6503-4051-aa04-228ad32b55fe.png">            |  <img width="358" alt="After" src="https://user-images.githubusercontent.com/17258145/235486581-df482e2d-1ed6-4886-ad3e-310944af7a28.png">         |

### Checklist



This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30396)
